### PR TITLE
Add get media info, media download and capture Info API

### PR DESCRIPTION
**Pull request by Sushma7785**

### CODE REVIEW

File: Yuneec_SDK_iOS/YNCSDKCamera.mm [hunk 1]
```diff
@@ -746,6 +746,48 @@ void receive_metering_result(YNCMeteringCompletion completion, Camera::Result re
...
+        NSError *error = nullptr;
+        if (result != Camera::Result::SUCCESS) {
+            NSString *message = [NSString stringWithFormat:@"%s", Camera::result_str(result)];
+            error = [[NSError alloc] initWithDomain:@"Camera"
+                                               code:(int)result
+                                           userInfo:@{@"message": message}];
+            completion(nil, error);
+        }
+        else {
+            NSMutableArray<YNCCameraMediaInfo *> *YNCCameraMediaInfos = [NSMutableArray new];
```
Comment (julianoes) [inline Yuneec_SDK_iOS/YNCSDKCamera.mm]: I have newer seen this allocation with `new`.
I would usually do `[[NSMutableArray alloc]init]`.

Also, I wouldn't make an array with pointers to `YNCCameraMediaInfo` but actual struct objects, and then just copy them in there by value.

Comment (Sushma7785) [reply Yuneec_SDK_iOS/YNCSDKCamera.mm]: Sorry forgot to change this! I am not sure what you mean. Can you please elaborate?

Comment (julianoes) [reply Yuneec_SDK_iOS/YNCSDKCamera.mm]: What I meant is that you don't need to allocate the `YNCCameraMediaInfo` on the heap using a pointer and `new` but, I think, you can just create it on the stack:
```
YNCCameraMediaInfo mediaInfo;
mediaInfo.path = @(mediaInfos[i].path.c_str());
mediaInfo.sizeMib = mediaInfos[i].size_mib;
```
and then add it (by value) so it gets copied.

Comment (julianoes) [reply Yuneec_SDK_iOS/YNCSDKCamera.mm]: Actually, it doesn't work as I suggested, so nevermind.

**Pull request merged**